manifests/init.pp

```diff
@@ -121,6 +144,59 @@
     fail('set_votequorum is true, but no quorum_members have been passed.')
   }
 
+  if $packages {
+    warning('$corosync::packages is deprecated, use $corosync::package_{corosync,pacemaker} instead!')
+
+    package{ $packages:
+      ensure => present,
+    }
+
+    # Ensure no options conflicting with $packages are set:
+
+    if $package_corosync {
+      fail('$corosync::package_corosync and $corosync::packages must not be mixed!')
+    }
+    if $package_pacemaker {
+      fail('$corosync::package_pacemaker and $corosync::packages must not be mixed!')
+    }
+    if $version_corosync {
+      fail('$corosync::version_corosync and $corosync::packages must not be mixed!')
+    }
+    if $version_pacemaker {
+      fail('$corosync::version_pacemaker and $corosync::packages must not be mixed!')
+    }
+  }
+    else {
+      # Handle defaults for new-style package parameters here to allow co-existence with $packages.
+      if ! $package_corosync {
+        $_package_corosync = true
+      }
+
+      if ! $package_pacemaker {
+        $_package_pacemaker = true
+      }
```

Comment:
Cool. Nice pattern. Will be glad to lose it after next major release TBH though ;-)

So defaults will work for `$_package_pacemaker`, but if `$package_pacemaker` does get a value, will that be handled at all? Will `$version_pacemaker`?